### PR TITLE
Create new BNB module

### PR DESCRIPTION
**Pull request by kxl4126**
This new bnb module can be used to gradually migrate naming from BSC to BNB. The old BSC module still exists in terms of dependencies that still rely on it. The goal is to eventually deprecate sdk-coin-bsc and use sdk-coin-bnb.

Will verify with an alpha release in a WP PR namespace, but this should no longer be a breaking change.

Dockerfile run commands have been consolidated to solve the error of `max depth exceeded`.

### CODE REVIEW

File: modules/statics/src/coins.ts [hunk 1]
```diff
@@ -953,6 +955,16 @@ export const coins = CoinMap.fromCoins([
...
+  account(
+    'd0d44124-c7e9-4214-96ae-fbc6856ee3c2',
+    'bnb',
+    'BNB Smart Chain',
+    Networks.main.bnb,
+    18,
+    UnderlyingAsset.BNB,
+    BaseUnit.BNB,
+    [...ETH_FEATURES_WITH_MMI, CoinFeature.TSS, CoinFeature.TSS_COLD, CoinFeature.EVM_WALLET]
+  ),
```
Comment (kxl4126) [inline modules/statics/src/coins.ts:967]: Should we use a unique uuidv4 for these "copied" tokens? Or should they get their own uuidv4

Comment (kxl4126) [reply modules/statics/src/coins.ts:967]: nevermind it forces a different uuid

**Review by zahin-mohammad — COMMENTED** [review]
Not ready for review, flushing from review Q

**Review by sachushaji — COMMENTED** [review]
Flush queue, re-request when ready

File: modules/account-lib/package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "@bitgo/sdk-coin-avaxc": "^2.4.18",
     "@bitgo/sdk-coin-avaxp": "^3.5.18",
     "@bitgo/sdk-coin-bld": "^1.1.0",
+    "@bitgo/sdk-coin-bnb": "^1.0.0",
```
Comment (mmcshinsky-bitgo) [inline modules/account-lib/package.json:35]: Add to tsconfig as well.

File: modules/bitgo/package.json [hunk 3]
```diff
@@ -56,6 +56,7 @@
     "@bitgo/sdk-coin-bch": "^1.3.27",
     "@bitgo/sdk-coin-bcha": "^1.5.22",
     "@bitgo/sdk-coin-bld": "^1.1.0",
+    "@bitgo/sdk-coin-bnb": "^1.0.0",
```
Comment (mmcshinsky-bitgo) [inline modules/bitgo/package.json:59]: Add to tsconfig as well

File: modules/sdk-coin-bnb/src/tbnb.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Testnet Bsc
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-bnb/src/tbnb.ts:2]:
```suggestion
 * Testnet Bnb
```

File: modules/sdk-coin-bnb/test/resources/bsc.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export const TEST_ACCOUNT = {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-bnb/test/resources/bsc.ts:1]: file name to bnb

**Review by mmcshinsky-bitgo — COMMENTED** [review]
Looks like a lot of bcs references and files to be renamed to bnb within the new module.

File: Dockerfile [hunk 6]
```diff
@@ -39,119 +39,122 @@ COPY --from=builder /tmp/bitgo/modules/express /var/bitgo-express/
...
 COPY --from=builder /tmp/bitgo/modules/sdk-coin-atom /var/modules/sdk-coin-atom/
-RUN cd /var/modules/sdk-coin-atom && yarn link
 COPY --from=builder /tmp/bitgo/modules/sdk-coin-avaxc /var/modules/sdk-coin-avaxc/
-RUN cd /var/modules/sdk-coin-avaxc && yarn link
 COPY --from=builder /tmp/bitgo/modules/sdk-coin-avaxp /var/modules/sdk-coin-avaxp/
-RUN cd /var/modules/sdk-coin-avaxp && yarn link
 COPY --from=builder /tmp/bitgo/modules/sdk-coin-eth /var/modules/sdk-coin-eth/
-RUN cd /var/modules/sdk-coin-eth && yarn link
 COPY --from=builder /tmp/bitgo/modules/sdk-coin-bld /var/modules/sdk-coin-bld/
-RUN cd /var/modules/sdk-coin-bld && yarn link
```
Comment (mmcshinsky-bitgo) [inline Dockerfile:80]: Was this intentional?

Comment (mmcshinsky-bitgo) [reply Dockerfile:80]: If so the script is going to either bork on this file, or possibly create duplicates. Recommend testing changes with script updates.

Comment (kxl4126) [reply Dockerfile:80]: Yes, I was running into a persistent `max depth exceeded` error on the docker-build step, which is caused by too many layers. Consolidating the run commands should fix this, but I'm not super experienced with docker to be sure the change should work the same.

File: modules/statics/src/account.ts [hunk 7]
```diff
@@ -887,6 +889,38 @@ export function bscToken(
...
+  id: string,
+  name: string,
+  fullName: string,
+  decimalPlaces: number,
+  contractAddress: string,
+  asset: UnderlyingAsset,
+  features: CoinFeature[] = AccountCoin.DEFAULT_FEATURES,
+  prefix = '',
+  suffix: string = name.toUpperCase(),
+  network: EthereumNetwork = Networks.main.bsc,
```
Comment (mmcshinsky-bitgo) [inline modules/statics/src/account.ts:902]: Should this be bnb?

File: Dockerfile (hunk 6 above)
Comment (kxl4126) [reply Dockerfile:80]: sorry, what do you mean by "script updates"?

**Review by mmcshinsky-bitgo — COMMENTED** [review]
lgtm. just a couple questions

File: Dockerfile (hunk 6 above)
Comment (mmcshinsky-bitgo) [reply Dockerfile:80]: There is a script that manages this file. `scripts/update-dockerfile`

Comment (kxl4126) [reply Dockerfile:80]: okay updated script to do the same thing

**Review by mmcshinsky-bitgo — DISMISSED** [review]

**Review by theMobiusStrip — DISMISSED** [review]

**Review by theMobiusStrip — APPROVED** [review]

**Pull request merged**